### ISSUE TITLE
`OneTimeKeys` is not defined by the CS API spec

### ISSUE THREAD

**Issue by DMRobertson** [matrix-org/matrix-spec#1287]
**Link to problem area**: https://spec.matrix.org/unstable/client-server-api/#post_matrixclientv3keysupload

**Issue**
`fallback_keys` and `one_time_keys` are defined as `OneTimeKeys`. This type is not defined in the client-server spec.

There are no mandatory keys for `OneTimeKeys`. I expect this confuses our go template langauge machinery.

https://github.com/matrix-org/matrix-spec/blob/dcc1f4adee2be41069045ddbf3ef913e9c19d116/data/api/client-server/definitions/one_time_keys.yaml#L1-L24

**Comment by DMRobertson** [comment]
Possibly related: #1162

**Comment by DMRobertson** [comment]
#1127 suggests that this can't be done unless we upgrade to OpenAPI v3.

**Comment by KitsuneRal** [comment]
Exactly; that's the only reason it's not defined. Actually, I think I even left a commented out definition in the API definition file...

**Comment by richvdh** [comment]
We now have OpenAPI v3, and this could be looked at again
